integration_tests/src/main/python/delta_lake_utils.py

```diff
@@ -157,12 +157,15 @@ def setup_delta_dest_table(spark, path, dest_table_func, use_cdf, partition_colu
     dest_df = dest_table_func(spark)
     writer = dest_df.write.format("delta")
     ddl = schema_to_ddl(spark, dest_df.schema)
-    table_properties = {}
-    if use_cdf:
-        table_properties['delta.enableChangeDataFeed'] = 'true'
-    if enable_deletion_vectors:
-        table_properties['delta.enableDeletionVectors'] = 'true'
-    if len(table_properties) > 0:
+    table_properties = {
+        'delta.enableChangeDataFeed': str(use_cdf).lower(),
+    }
+    # prevent on 11.3: Unknown configuration was specified: delta.enableDeletionVectors
+    # TODO OSS Delta 2.3+
+    if is_databricks122_or_later():
+        table_properties['delta.enableDeletionVectors'] = str(enable_deletion_vectors).lower()
+
+    if use_cdf or enable_deletion_vectors:
```

Comment:
What happens if we enable deletion_vectors and use_cdf is False, but we are not on databrick 122 or later? I know that this didn't really change from the previous code. It is just confusing to me and I would like to understand better what is happening. 

nit: Also why is `writer.mode("append")` in a separate if clause right after this one?

Reply:
We cannot enable_deleltion_vectors before 12.2, hence all tests where enable_deletion_vectors may be True are 
```
@pytest.mark.skipif(not supports_delta_lake_deletion_vectors(), reason="Deletion vectors are new in Spark 3.4.0 / DBR 12.2")
```
re: write.mode("append"). I did not want to introduce changes beyond the absolute minimum.